app/controllers/users_controller.rb

```diff
@@ -1,6 +1,7 @@
 class UsersController < ApplicationController
+  
   before_action :set_user, only: %i[ show liked feed followers following discover ]
-
+  before_action {authorize @user || User}
```

Comment:
We don't really need `|| User` here as there is a `@user` record for each of our action.